src/test/java/com/puppycrawl/tools/checkstyle/utils/TokenUtilTest.java

```diff
@@ -395,4 +395,212 @@ public void testIsBooleanLiteralType() {
+
+        assertWithMessage("Token type did not match")
+            .that(result1)
+            .isTrue();
+        assertWithMessage("Token type did not match")
+            .that(result2)
+            .isTrue();
+    }
+
+    @Test
+    public void testIsOfTypeFirstFalse() {
+        final int type = TokenTypes.LITERAL_CATCH;
+        final DetailAstImpl astForTest1 = new DetailAstImpl();
+        final DetailAstImpl astForTest2 = null;
+        astForTest1.setType(type);
+        final boolean result1 = TokenUtil.isOfType(type, TokenTypes.LITERAL_FOR);
+        final boolean result2 = TokenUtil.isOfType(astForTest1, TokenTypes.LITERAL_FOR);
+        final boolean result3 = TokenUtil.isOfType(astForTest2, TokenTypes.LITERAL_FOR);
+
+        assertWithMessage("Token type should not match")
+            .that(result1)
+            .isFalse();
+        assertWithMessage("Token type should not match")
+            .that(result2)
+            .isFalse();
+        assertWithMessage("Token type should not match")
+            .that(result3)
+            .isFalse();
+    }
+
+    @Test
+    public void testIsOfTypeSecondTrue() {
+        final int type = TokenTypes.LITERAL_CATCH;
+        final DetailAstImpl astForTest = new DetailAstImpl();
+        astForTest.setType(type);
+        final boolean result1 = TokenUtil.isOfType(type, TokenTypes.LITERAL_IF,
+                                                   TokenTypes.LITERAL_CATCH);
+        final boolean result2 = TokenUtil.isOfType(astForTest, TokenTypes.LITERAL_IF,
+                                                   TokenTypes.LITERAL_CATCH);
+
+        assertWithMessage("Token type did not match")
+            .that(result1)
+            .isTrue();
+        assertWithMessage("Token type did not match")
+            .that(result2)
+            .isTrue();
+    }
+
+    @Test
+    public void testIsOfTypeSecondFalse() {
+        final int type = TokenTypes.LITERAL_CATCH;
+        final DetailAstImpl astForTest1 = new DetailAstImpl();
+        final DetailAstImpl astForTest2 = null;
+        astForTest1.setType(type);
+        final boolean result1 = TokenUtil.isOfType(type, TokenTypes.LITERAL_IF,
+                                                   TokenTypes.LITERAL_ELSE);
+        final boolean result2 = TokenUtil.isOfType(astForTest1, TokenTypes.LITERAL_IF,
+                                                   TokenTypes.LITERAL_ELSE);
+        final boolean result3 = TokenUtil.isOfType(astForTest2, TokenTypes.LITERAL_IF,
+                                                   TokenTypes.LITERAL_ELSE);
+
+        assertWithMessage("Token type should not match")
+            .that(result1)
+            .isFalse();
+        assertWithMessage("Token type should not match")
+            .that(result2)
+            .isFalse();
+        assertWithMessage("Token type should not match")
+            .that(result3)
+            .isFalse();
+    }
+
+    @Test
+    public void testIsOfTypeThreeTrue() {
```

Comment:
incorrect name, you're testing `isOfType` with four type arguments here

Reply:
Changes Done